File: frontend/src/types/sector.ts
export interface SectorInstrument {
  symbol: string;
  name: string;
  change_percent: number | null;
  market_cap: number | null;
}

export interface SectorItem {
  sector_name: string;
  sector_name_ko: string | null;
  weighted_change_percent: number | null;
  total_market_cap: number | null;
  strongest_symbol: string | null;
  weakest_symbol: string | null;
  instruments: SectorInstrument[];
}

export interface HeatmapResponse {
  KOR: SectorItem[];
  US: SectorItem[];
  top_sectors: string[];
  bottom_sectors: string[];
}

File: backend/app/schemas/sector.py
from pydantic import BaseModel


class SectorItem(BaseModel):
    sector_name: str
    sector_name_ko: str | None
    weighted_change_percent: float | None
    total_market_cap: float | None
    strongest_symbol: str | None
    weakest_symbol: str | None
    instruments: list[dict]   # {symbol, name, change_percent, market_cap}


class HeatmapResponse(BaseModel):
    KOR: list[SectorItem]
    US: list[SectorItem]
    top_sectors: list[str]
    bottom_sectors: list[str]
